The buffer pointed to by "packet" won't be initialized if k5_retcode is
not zero, so we shouldn't try to free it.

--- cyrus-sasl-2.1.21/saslauthd/auth_krb5.c	2005-09-16 18:12:24.000000000 -0400
+++ cyrus-sasl-2.1.21/saslauthd/auth_krb5.c	2005-09-16 18:12:26.000000000 -0400
@@ -322,7 +322,9 @@
     /* all is good now */
     result = 1;
  fini:
-    krb5_free_data_contents(context, &packet);
+    if (!k5_retcode) {
+        krb5_free_data_contents(context, &packet);
+    }
     krb5_free_principal(context, server);
     
     return result;
